pkgs/native_assets_builder/lib/src/package_layout/package_layout.dart

```diff
@@ -29,26 +30,30 @@ class PackageLayout {
 
   final Uri packageConfigUri;
 
-  PackageLayout._(
-      this.rootPackageRoot, this.packageConfig, this.packageConfigUri);
+  PackageLayout._(this._fileSystem, this.rootPackageRoot, this.packageConfig,
```

Comment:
We could consider using `Directory rootPackageRoot` which would encapsulate both the `Uri` and the `_fileSystem`.